### PR TITLE
Swagger refactor/cleanup

### PR DESCRIPTION
**Pull request by jwhonce**
* Remove duplicate or unused types and constants
* Move all documetation-only models and responses into swagger package
* Remove all unecessary names, go-swagger will determine names from
  struct declarations
* Use Libpod suffix to differentiate between compat and libpod models
  and responses. Taken from swagger:operation declarations.
* Models and responses that start with lowercase are for swagger use
  only while uppercase are used "as is" in the code and swagger comments
* Used gofumpt on new code

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

[NO NEW TESTS NEEDED]

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: pkg/api/handlers/compat/images_build.go [hunk 1]
```diff
@@ -790,7 +790,7 @@ func extractTarFile(r *http.Request) (string, error) {
 	}
 
 	buildDir := filepath.Join(anchorDir, "build")
-	err = os.Mkdir(buildDir, 0700)
+	err = os.Mkdir(buildDir, 0o700)
```
Comment (TomSweeneyRedHat) [inline pkg/api/handlers/compat/images_build.go:793]: I know this kind of notation is happy in python, is it good in Go too?

Comment (jwhonce) [reply pkg/api/handlers/compat/images_build.go:793]: @TomSweeneyRedHat yes, gofumpt makes the change automatically as well as other clean-ups.

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM
assuming happy tests, one question for my education

**Pull request merged**